devops/azure-pipelines/maven-build-test-job-template.yml

```diff
@@ -0,0 +1,50 @@
+parameters:
+  - name: 'jobName'
+    type: string
+  - name: 'serviceConnection'
+    type: string
+  - name: 'organization'
+    type: string
+  - name: 'testPath' # file path to test results
+    type: string
+jobs:
+  - job: ${{ parameters.jobName }}
+    steps:
+      - task: SonarCloudPrepare@1
+        condition: ne(variables['Build.Reason'], 'PullRequest')
+        inputs:
+          SonarCloud: 'Forge-SonarCloud'
+          organization: '1058-sophia-gavrila-forge'
+          scannerMode: 'Other'
+          extraProperties: |
+           sonar.branch.name=$(Build.SourceBranchName) 
+           sonar.projectKey=1058-Sophia-Gavrila-Forge_Forge-Backend
+           sonar.projectName=Forge-Backend
+      - task: SonarCloudPrepare@1
+        condition: eq(variables['Build.Reason'], 'PullRequest')
+        inputs:
+          SonarCloud: 'Forge-SonarCloud'
+          organization: '1058-sophia-gavrila-forge'
+          scannerMode: 'Other'
+          extraProperties: |
+           sonar.projectKey=1058-Sophia-Gavrila-Forge_Forge-Backend
+           sonar.projectName=Forge-Backend
```

Comment:
avoid you could use conditional insertion with template syntax to avoid this duplication, but it's just not necessary because sonar will auto detect the git stuff.

Reply:
While I agree that the sonar.projectKey probably isn't necessary, the sonar.projectName is overwriting the name that sonar pulls from the pom file, demo, with something more descriptive to the project it belongs to, Forge-Backend.

Reply:
@alexanderjervis , I think the conditional insertion thing Nick is talking about is about sonar.branch.name, and how we had to put it in for normal sonar runs to work, and take it out for PR sonar runs to work... Maybe we're missing something and being a cargo cult right now, I'll try to look more into it.

Reply:
yes that's what i meant. but like i said it just shouldn't be necessary in the first place since sonar scanner is supposed to detect the whole git situation at runtime just by looking at the .git folder, github.com, etc. it's working for other repos as far as i'm aware, so look at the sonar step logs on those repos for the PR and non PR builds and compare with what you get here.
though we're running out of time so it may be better to keep this kluge and just give it enough comments for future people to evaluate it.